test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -159,6 +160,48 @@ void performFailoverTest() {
         log.info("Read count: {}, write count: {} with Segment Store  and controller failover after sleep", currentReadCount1, currentWriteCount1);
     }
 
+    void performFailoverForTestsInvolvingTxns() {
+
+        log.info("Test with 3 controller, segment store instances running and without a failover scenario");
+        log.info("Read count: {}, write count: {} without any failover",
+                testState.eventReadCount.get(), testState.eventWriteCount.get());
+
+        //check reads and writes after sleeps
+        log.info("Sleeping for {} ", WAIT_AFTER_FAILOVER_MILLIS);
+        Exceptions.handleInterrupted(() -> Thread.sleep(WAIT_AFTER_FAILOVER_MILLIS));
+        log.info("Read count: {}, write count: {} without any failover after sleep before scaling",
+                testState.eventReadCount.get(),  testState.eventWriteCount.get());
+
+        //Scale down segment store instances to 2
+        segmentStoreInstance.scaleService(2, true);
+        //zookeeper will take about 30 seconds to detect that the node has gone down
+        Exceptions.handleInterrupted(() -> Thread.sleep(WAIT_AFTER_FAILOVER_MILLIS));
```

Comment:
without txn pings, what happens in these 30 seconds.. do transactions timeout?

Reply:
I have picked up a txn here. we can see that txn sets it status to `COMMITTING`  after segmentstore failover and gets committed while controller failover is happening.
this is from the test logs, where test failed because 2 transactions failed to commit, and both have failed after all failovers.
```
2017-10-14 15:29:21,131 81111 [ReadTxnWriteScaleWithFailoverTest-main-7] DEBUG i.p.t.system.AbstractFailoverTests - Txn id 64f15db8-0766-4a6e-9bc3-2c20d6f0e9be status is COMMITTING
2017-10-14 15:29:22,922 82902 [Time-limited test] INFO  i.p.t.system.AbstractFailoverTests - Read count: 31700, write count: 30600 after Segment Store  failover after sleep
2017-10-14 15:30:03,023 123003 [Time-limited test] INFO  i.p.t.system.AbstractFailoverTests - Read count: 48950, write count: 47600 after controller failover after sleep
2017-10-14 15:29:30,825 90805 [ReadTxnWriteScaleWithFailoverTest-main-7] DEBUG i.p.t.system.AbstractFailoverTests - Txn id 64f15db8-0766-4a6e-9bc3-2c20d6f0e9be status is COMMITTED
```